### PR TITLE
add norm1d and 2d support

### PR DESCRIPTION
**Pull request by suiyuan2009**
#2826
- add norm1d support
- I'll write norm2d, and add test

### CODE REVIEW

**Comment by vrv** [comment]
I'm not sure how much of this CL overlaps (if any) with the existing ops, assigning to those who are more familiar

**Comment by suiyuan2009** [comment]
@vrv I see #2826 , so I think it's not implemented before

File: tensorflow/core/kernels/norm_ops.cc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+namespace functor {
+
+template <typename T>
+struct Norm1D<CPUDevice, T> {
+   void operator()(const CPUDevice& d, typename TTypes<T>::ConstFlat x,
+		   typename TTypes<T>::ConstFlat mean,
+		   typename TTypes<T>::ConstFlat stdd,
+		   typename TTypes<T>::Flat out) {
+	   auto new_x = x - mean;
+	   auto z = - new_x.square() / (stdd.constant(2.0) * stdd.square());
```
Comment (benoitsteiner) [inline tensorflow/core/kernels/norm_ops.cc]: I would force the 2.0 constant to be of the same type as T by writing the code as:
   stdd.constant(static_cast<T>(2.0))

File: tensorflow/core/kernels/norm_ops.cc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace functor {
+
+template <typename T>
+struct Norm1D<CPUDevice, T> {
+   void operator()(const CPUDevice& d, typename TTypes<T>::ConstFlat x,
+		   typename TTypes<T>::ConstFlat mean,
+		   typename TTypes<T>::ConstFlat stdd,
+		   typename TTypes<T>::Flat out) {
+	   auto new_x = x - mean;
```
Comment (benoitsteiner) [inline tensorflow/core/kernels/norm_ops.cc]: you could define new_x as mean-x: this would avoid the need for the negation in the computation of z below

**Comment by ebrevdo** [comment]
This is a premature optimization. See tf.contrib.distributions.

This op doesn't support broadcasting and has no registered gradients.  It's not clear that it's any faster the equivalent tensorflow code.

**Comment by suiyuan2009** [comment]
@benoitsteiner @ebrevdo thank you for the code review, learned  a lot,  I'll close this pr.

**Comment by suiyuan2009** [comment]
hope I can help if optimization in tf.contrib is needed

File: tensorflow/core/kernels/norm_ops.cc (hunk 2 above)
Comment (suiyuan2009) [inline tensorflow/core/kernels/norm_ops.cc]: thank you anyway, but negation can't be avoided becase `square()`

**Comment by vrv** [comment]
I think some numbers about why tf.contrib.distributions is not fast enough would be needed before accepting this (in general we only try to add custom ops when we can't get the equivalent from compositions of primitive ops at a reasonable speed).

**Pull request closed without merging**